api/pkg/service/resource/resource.go

```diff
@@ -346,6 +347,43 @@ func filterByKind(t string) func(db *gorm.DB) *gorm.DB {
 	}
 }
 
+func filterByKinds(t []string) func(db *gorm.DB) *gorm.DB {
+	if len(t) == 0 {
+		return noop
+	}
+
+	return func(db *gorm.DB) *gorm.DB {
+		t = lower(t)
+		return db.Where("LOWER(kind) IN (?)", t)
+	}
+}
+
+func filterByTags(t []string) func(db *gorm.DB) *gorm.DB {
+	if t == nil {
+		return noop
+	}
+
+	return func(db *gorm.DB) *gorm.DB {
+		t = lower(t)
+
+		resID := getResourceIDs(t, db)
+
+		return db.Where("id IN (?)", resID)
+	}
+}
+
+func getResourceIDs(t []string, db *gorm.DB) (resID []uint) {
+
+	query := db.Model(&model.Resource{}).Select("DISTINCT(resources.id)").
```

Comment:
Is `Select("DISTINCT(resources.id)")` requires now if we are using pluck ?

Reply:
Using pluck it selects a single column from the table it does not select unique id's from the table

Reply:
`Pluck("DISTINCT(resources.id)", &resID)` will this work?

Reply:
Yes this will work !